client/src/components/File.js

```diff
@@ -2,225 +2,154 @@
 
 import React from 'react';
 import TransitionGroup from "react-addons-css-transition-group";
-import { getHumanReadableBytes } from '../utils/utils.js';
-import apiurl from 'utils/apiurl';
-import 'styles/File.scss';
-import Highlight from 'components/plugins/highlight';
+import Clipboard from 'clipboard';
 import highlight from 'highlight.js'
+import Highlight from 'components/plugins/highlight';
+import { getHumanReadableBytes } from '../utils/utils.js';
 import ChannelActions from 'actions/ChannelActions';
-import Clipboard from 'clipboard';
-
 import MessageStore from 'stores/MessageStore';
-
+import 'styles/File.scss';
 
 import Logger from 'logplease';
 const logger = Logger.create('Clipboard', { color: Logger.Colors.Magenta });
 
-var getFileUrl = apiurl.getApiUrl() + "/api/cat/";
+function readUTF8String(bytes) {
```

Comment:
You are of course right. I'll rework what i did there, also wrt to video streaming etc. js-ipfs cat of course gives me a readable stream that emits chunks that are Buffers, which have a .toString method which handles UTF-8 just fine.